flare/algorithm_zoo/ddpg.py

```diff
@@ -0,0 +1,168 @@
+from copy import deepcopy
+import math
+from flare.framework.algorithm import Algorithm, Model
+from flare.framework import common_functions as comf
+import torch
+import torch.nn as nn
+import torch.optim as optim
+
+
+def fanin_init(m):
+    fan_in, _ = nn.init._calculate_fan_in_and_fan_out(m.weight)
+    bound = 1.0 / math.sqrt(fan_in)
+    nn.init.uniform_(m.weight, -bound, bound)
+    nn.init.uniform_(m.bias, -bound, bound)
+
+
+def uniform_init(m, bound):
+    nn.init.uniform_(m.weight, -bound, bound)
+    nn.init.uniform_(m.bias, -bound, bound)
+
+
+class ContinuousDeterministic(Model):
+    """
+    Model used by DDPG.
+    """
+
+    def __init__(self, input_dims, action_dims):
+        super(ContinuousDeterministic, self).__init__()
+        self.input_dims = input_dims
+        self.action_dims = action_dims
+
+        self.policy_fc1 = nn.Linear(input_dims, 400)
```

Comment:
Probably consider generalizing the hidden dims of the layers by passing them as arguments of the class.